dlrover/python/elastic_agent/torch/ckpt_saver.py

```diff
@@ -908,6 +947,39 @@ def commit_checkpoint(  # type: ignore
             time.sleep(2)
 
 
+class DeepSpeedCheckpointSaver(AsyncCheckpointSaver):
+    @classmethod
+    def get_checkpoint_tracker_filename(cls, checkpoint_dir):
+        """
+        Get the path of tracker file to record the latest checkpointing
+        step.
+
+        Args:
+            checkpoint_dir (str): the checkpoint directory.
+
+        Returns:
+            str: the path of tracker file.
+        """
+        fname = "latest"
+        return os.path.join(checkpoint_dir, fname)
+
+    def persist_to_storage(  # type: ignore
+        self, state_dict, ckpt_config: DeepSpeedCheckpointConfig
+    ):
+        """Persist the checkpoint from CPU memory buffer into the storage."""
+        model_sd = state_dict.get(CheckpointConstant.MODEL_STATES_NAME, {})
```

Comment:
Deepspeed has its own checkpoint saving and loading function[https://deepspeed.readthedocs.io/en/latest/model-checkpointing.html](url). Is the  `persist_to_storage ` compatible to  `Deepspeed `save load function. What if  the `state_dict` requires cuda memory larger than a single gpu could hold?

Reply:
Yes, It is compatible with `DeepSpeed.save/load`. The state dict and path of the method are generated by `DeepSpeed.save/load`. You can see the detail in https://github.com/intelligent-machine-learning/dlrover/blob/cfbfaf652bb57f742f5932a7040adc0641a9e83c/dlrover/trainer/torch/megatron/async_checkpoint.py#L77-L113 which will be submitted in the next PR.